lib/phlex/context.rb

```diff
@@ -6,24 +6,59 @@ def initialize(user_context: {}, view_context: nil)
 		@buffer = +""
 		@capturing = false
 		@user_context = user_context
+		@fragments = nil
+		@fragment_depth = 0
+		@halt_signal = nil
 		@view_context = view_context
 	end
 
 	attr_accessor :buffer, :capturing, :user_context
 
-	attr_reader :view_context
+	attr_reader :fragments, :fragment_depth, :view_context
+
+	def target_fragments(fragments)
+		@fragments = fragments.to_h { |it| [it, true].freeze }
+	end
+
+	def around_render
+		return yield if !@fragments || @halt_signal
+
+		catch do |signal|
+			@halt_signal = signal
+			yield
+		end
+	end
+
+	def in_target_fragment
+		!@fragments || @fragment_depth > 0
+	end
+
+	def begin_target(id)
+		@fragment_depth += 1 if @fragments&.[](id)
+	end
+
+	def end_target(id)
+		return unless @fragments&.[](id)
+
+		@fragments.delete(id)
+		@fragment_depth -= 1
+		throw @halt_signal if @fragments.length == 0
```

Comment:
This is how we had it before, but throwing `nil` is odd here. I think we can rework this a bit to make it better.